src/gui/MainWindow.cc

```diff
@@ -1314,7 +1315,24 @@ void MainWindow::compileEnded()
   clearCurrentOutput();
   GuiLocker::unlock();
   if (designActionAutoReload->isChecked()) autoReloadTimer->start();
+#ifdef ENABLE_GUI_TESTS
+  emit compilationDone(this->rootFile);
+#endif
+}
+
+#ifdef ENABLE_GUI_TESTS
+std::shared_ptr<AbstractNode> MainWindow::instantiateRootFromSource(SourceFile* file)
+{
+    EvaluationSession session{file->getFullpath()};
+    ContextHandle<BuiltinContext> builtin_context{Context::create<BuiltinContext>(&session)};
+    setRenderVariables(builtin_context);
+
+    std::shared_ptr<const FileContext> file_context;
+    std::shared_ptr<AbstractNode> node = this->rootFile->instantiate(*builtin_context, &file_context);
+
+    return node;
 }
```

Comment:
I had to add this function so I can test the testMCAD in moduleCache. Maybe an equivalent test can be done without touching the API.

Reply:
Let's keep it for now